### PR TITLE
feat(cloudfront): responseHttpStatus defaults to httpStatus in errorResponses

### PR DESCRIPTION
**Pull request by jogold**
Make `responseHttpStatus` default to `httpStatus` if `responsePagePath`
is defined.

This avoids repeating the error code and is more in line with the actual
`@default` documentation (which actually led me to think that it
already defaulted to `httpStatus`).

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
Not sure this requires a README update.

File: packages/@aws-cdk/aws-cloudfront/lib/distribution.ts [hunk 1]
```diff
@@ -393,21 +393,18 @@ export class Distribution extends Resource implements IDistribution {
...
-      }
-      if (!errorResponse.responseHttpStatus && !errorResponse.ttl) {
-        throw new Error('A custom error response without either a \'responseCode\' or \'errorCachingMinTtl\' is not valid.');
-      }
-    }
-    this.errorResponses.forEach(e => validateCustomErrorResponse(e));
 
     return this.errorResponses.map(errorConfig => {
+      if (!errorConfig.responseHttpStatus && !errorConfig.ttl && !errorConfig.responsePagePath) {
+        throw new Error('A custom error response without either a \'responseHttpStatus\' or \'ttl\' is not valid.');
```
Comment (njlynch) [inline packages/@aws-cdk/aws-cloudfront/lib/distribution.ts:399]: Can you update this error message to be more inline with the new check?

**Review by njlynch — COMMENTED** [review]
Looks good! Based on the docstring, it's possible this is how I originally intended it, then got distracted somewhere. :) Just one minor update.

And I agree, no need for a README update for this.

**Review by njlynch — APPROVED** [review]
Awesome, thanks!

**Pull request merged**